pkg/orchestrator/scheduler/batch_service_job.go

```diff
@@ -162,8 +187,23 @@ func (b *BatchServiceJobScheduler) placeExecs(ctx context.Context, execs execSet
 	return nil
 }
 
+func (b *BatchServiceJobScheduler) handleRetry(plan *models.Plan, job *models.Job, failures int) {
+	// delay = RetryDelay * RetryDelayGrowthFactor ^ failures
+	// ...but never more than MaximumRetryDelay
+	delay := gomath.Min(
+		float64(job.RetryDelay)*gomath.Pow(job.RetryDelayGrowthFactor, float64(failures)),
+		float64(job.MaximumRetryDelay))
+
```

Comment:
exponential backoff usually multiplies the backoff duration with the growth factor instead of using powers

Reply:
Yeah, that's what I started with, but I couldn't see a particularly nice place to store the current backoff duration. Mutating the job didn't seem to fit with the philosophy of the system, which seems to avoid mutating the job (there's only a few, limited, kinds of updates accepted by the Job Store), so I felt that using exponentiation to re-compute the result of those repeated multiplications from scratch each time was more in keeping with how the rest of the scheduler worked, purely by considering the history of past executions.